### ISSUE TITLE
Python3 compatibility

### ISSUE THREAD

**Issue by jamesacampbell** [cpbotha/nvpy#153]
What is the level of effort and scope to get Python3 working? I would love to use this tool, but I really don't want to use 2.7, as it really seems like a step backwards. If there is a list of tasks to be done, I could possible help with some forks and pull requests.

**Comment by ethet** [comment]
I have a forked version of nvPY running on Python 3.7 [over here](https://github.com/ethet/nvpy). It is passing all of the tests and seems to be working well, though I would love some help in kicking the tires before I make a pull request (if a pull request is even desired).

You can run it from within the top level directory with:
`python3 -m nvpy`

**Comment by stefanv** [comment]
This has now become a blocker for Debian:

<details>

```
Package: src:nvpy
Version: 1.0.0+git20171203.c91062c-1
Severity: normal
Tags: sid bullseye
User: debian-python@lists.debian.org
Usertags: py2removal

Python2 becomes end-of-live upstream, and Debian aims to remove
Python2 from the distribution, as discussed in
https://lists.debian.org/debian-python/2019/07/msg00080.html

Your package either build-depends, depends on Python2, or uses Python2
in the autopkg tests.  Please stop using Python2, and fix this issue
by one of the following actions.

- Convert your Package to Python3. This is the preferred option.  In
  case you are providing a Python module foo, please consider dropping
  the python-foo package, and only build a python3-foo package.  Please
  don't drop Python2 modules, which still have reverse dependencies,
  just document them.
  
  This is the preferred option.

- If the package is dead upstream, cannot be converted or maintained
  in Debian, it should be removed from the distribution.  If the
  package still has reverse dependencies, raise the severity to
  "serious" and document the reverse dependencies with the BTS affects
  command.  If the package has no reverse dependencies, confirm that
  the package can be removed, reassign this issue to ftp.debian.org,
  make sure that the bug priority is set to normal and retitle the
  issue to "RM: PKG -- removal triggered by the Python2 removal".

- If the package has still many users (popcon >= 300), or is needed to
  build another package which cannot be removed, document that by
  adding the "py2keep" user tag (not replacing the py2remove tag),
  using the debian-python@lists.debian.org user.  Also any
  dependencies on an unversioned python package (python, python-dev)
  must not be used, same with the python shebang.  These have to be
  replaced by python2/python2.7 dependencies and shebang.

  This is the least preferred option.

If the conversion or removal needs action on another package first,
please document the blocking by using the BTS affects command, like

  affects <bug number of blocking py2removal bug> + src:nvpy

If there is no py2removal bug for that reverse-dependency, please file
a bug on this package (similar to this bug report).

If there are questions, please refer to the wiki page for the removal:
https://wiki.debian.org/Python/2Removal, or ask for help on IRC
#debian-python, or the debian-python@lists.debian.org mailing list.
```

</details>

**Comment by lfini** [comment]
A few months ago I've ported the then current version of nvpy code to
python3. I've been using it since then with no problem on my ubuntu laptop.

If you believe it  useful, I can upload it to github (in ten days or so,
because I'm on vacation right now)

Il ven 30 ago 2019, 14:54 Stefan van der Walt <notifications@github.com> ha
scritto:

**Comment by yuuki0xff** [comment]
@stefanv Sorry for the inconvenience.
I porting nvpy to python3.  The stable version is expected to be released in 1 to 2 weeks.

**Comment by stefanv** [comment]
Ah, I certainly did not mean it as a demand! I just wanted to bring it to your attention, but thank you for the quick response.

Once the port is ready, I will fix the Debian package accordingly.

**Comment by yuuki0xff** [comment]
@lfini  Thanks for porting.  I seems that your modifications are published on [ifini/nvpy](https://github.com/lfini/nvpy).  Are there any unpublished modifications in it?

I will merge from ifini/nvpy to master If it is no problem,

**Comment by lfini** [comment]
No, everything is in the repository. I just forgot I had already pushed it

Il ven 30 ago 2019, 17:33 yuuki <notifications@github.com> ha scritto:

**Comment by yuuki0xff** [comment]
Released v2.0.0.
